Zend/zend_multiply.h

```diff
@@ -19,10 +19,32 @@
 
 /* $Id$ */
 
+#include "zend_portability.h"
+
 #ifndef ZEND_MULTIPLY_H
 #define ZEND_MULTIPLY_H
 
-#if (defined(__i386__) || defined(__x86_64__)) && defined(__GNUC__)
+#if __has_builtin(__builtin_smull_overflow) && SIZEOF_LONG == SIZEOF_ZEND_LONG
```

Comment:
Ah, I see. Tricky.

So we'd need to add checks for `__builtin_saddl`(`l`)`_overflow`, `__builtin_ssubl`(`l`)`_overflow`, `__builtin_smull`(`l`)`_overflow` and define macros if they're available.

Okay, I can probably do that.